infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrackService.kt

```diff
@@ -442,11 +443,84 @@ class LocationTrackService(
         return lines.map { line -> line to alignments.getValue(line.getAlignmentVersionOrThrow()) }
     }
 
+    fun fillTrackAddress(splitPoint:SplitPoint, geocodingContext: GeocodingContext): SplitPoint {
+        val address = geocodingContext.getAddress(splitPoint.location)?.first
+        return when (splitPoint) {
+            is SwitchSplitPoint -> splitPoint.copy( address = address)
+            is EndpointSplitPoint -> splitPoint.copy( address = address)
+        }
+    }
+
+    fun fillTrackAddresses(duplicates: List<LocationTrackDuplicate>, geocodingContext:GeocodingContext) : List<LocationTrackDuplicate> {
+        return duplicates.map { duplicate ->
+            duplicate.copy(
+                duplicateStatus = duplicate.duplicateStatus.copy(
+                    startSplitPoint = duplicate.duplicateStatus.startSplitPoint?.let { splitPoint ->
+                        fillTrackAddress(splitPoint, geocodingContext)
+                        },
+                    endSplitPoint = duplicate.duplicateStatus.endSplitPoint?.let { splitPoint ->
+                        fillTrackAddress(splitPoint, geocodingContext)
+                    }
+                )
+            )
+        }
+    }
+
     @Transactional(readOnly = true)
     fun getInfoboxExtras(layoutContext: LayoutContext, id: IntId<LocationTrack>): LocationTrackInfoboxExtras? {
         return getWithAlignment(layoutContext, id)?.let { (locationTrack, alignment) ->
+            val start = alignment.start ?: return null
+            val end = alignment.end ?: return null
+            val geocodingContext = geocodingService.getGeocodingContext(layoutContext, locationTrack.trackNumberId)
+                ?: return null
+
             val duplicateOf = getDuplicateTrackParent(layoutContext, locationTrack)
-            val sortedDuplicates = getLocationTrackDuplicates(layoutContext, locationTrack, alignment)
+            val duplicates = fillTrackAddresses(
+                getLocationTrackDuplicates(layoutContext, locationTrack, alignment),
+                geocodingContext
+            )
+            val sortedDuplicates = duplicates.sortedBy { duplicate ->
+                duplicate.duplicateStatus.startSplitPoint?.address
+            }
+
+            val startAddress = geocodingContext?.getAddress(start)?.first
+            val startSwitchId = alignment.segments.firstOrNull()?.switchId as IntId?
+                ?: locationTrack.topologyStartSwitch?.switchId
+            val startSwitchSplitPoint = startSwitchId?.let { switchId ->
+                    SwitchSplitPoint(
+                        start,
+                        startAddress,
+                        switchId,
+                        JointNumber(0)
+                    )
+            }
+            val startSplitPoint = startSwitchSplitPoint
```

Comment:
Tämän saattaisi saada helpommin luettavaan muotoon jos sen muotoilisi jotenkin näin:

```
val startSplitPoint = if (startSwitchId != null) SwitchSplitPoint(...) else EndpointSplitPoint(...)
```

Ja sama tuossa alempana tuon `endSplitPoint`:in kanssa

Reply:
Paljonkin parempi...